homie-device/src/types.rs

```diff
@@ -58,33 +100,181 @@ impl Property {
         datatype: Datatype,
         settable: bool,
         unit: Option<&str>,
-        format: Option<&str>,
+        format: Option<String>,
```

Comment:
This is a bit inconsistent. Is the intention to make this function private?

Reply:
Fair point. I was thinking to keep this function public, in case anyone wants a generic way to create properties. So there are two options that seem plausible to me:
1. Add a private `Property::make` method or something like that which does what this function does, and have everything else call that. Change the signature of `Property::new` back to what it used to be.
2. Change `Property::new` to use `impl Into<String>` for all relevant parameters. The problem with this is that passing `None` as an `Option< impl Into<String>>` can't have its type inferred, so callers have to explicitly choose something (and it doesn't matter what).

Any idea which is more idiomatic?

Reply:
Probably 1. When in doubt, public functions should accept &str and do the clone themselves. Maybe call it from_fields() or something?

Reply:
Done.